glean-core/src/error_recording.rs

```diff
@@ -79,7 +81,7 @@ pub fn record_error(
     });
 
     log::warn!("{}: {}", identifier, message);
-    metric.add(glean, 1);
+    metric.add(glean, num_errors.into().unwrap_or(1) as i32);
```

Comment:
Why the back-and-forth-casting? We could just take `O: Into<Option<i32>>`.
Call sites with just a number should be fine.

Reply:
True, but it doesn't really make sense to accept "-1", right? I feel like we should really audit all our function signatures to make sure we use the proper types, to avoid surprises. I'm happy to file a bug about this. I'd change this as you suggested for now. @badboy thoughts?

Reply:
It does not make sense indeed. It's internal API, we could add a `debug_assert!(i > 0)`.

I filed https://bugzilla.mozilla.org/show_bug.cgi?id=1568526 for looking into this.